packages/nuxt/src/runtime/plugins/sentry-cloudflare.server.ts

```diff
@@ -0,0 +1,132 @@
+const TRACE_DATA_KEY = '__sentryTraceData';
+
+/**
+ * Sentry Cloudflare Nitro plugin for when using the "cloudflare-pages" preset in Nuxt.
+ * This plugin automatically sets up Sentry error monitoring and performance tracking for Cloudflare Pages projects.
+ *
+ * Instead of adding a `sentry.server.config.ts` file, export this plugin in the `server/plugins` directory
+ * with the necessary Sentry options to enable Sentry for your Cloudflare Pages project.
+ *
+ *
+ * @example Basic usage
+ * ```ts
+ * // nitro/plugins/sentry.ts
+ * import { defineNitroPlugin } from '#imports'
+ * import { sentryCloudflareNitroPlugin } from '@sentry/nuxt/module/plugins'
+ *
+ * export default defineNitroPlugin(sentryCloudflareNitroPlugin({
+ *   dsn: 'https://examplePublicKey@o0.ingest.sentry.io/0',
+ *   tracesSampleRate: 1.0,
+ * }));
+ * ```
+ *
+ * @example Dynamic configuration with nitroApp
+ * ```ts
+ * // nitro/plugins/sentry.ts
+ * import { defineNitroPlugin } from '#imports'
+ * import { sentryCloudflareNitroPlugin } from '@sentry/nuxt/module/plugins'
+ *
+ * export default defineNitroPlugin(sentryCloudflareNitroPlugin(nitroApp => ({
+ *   dsn: 'https://examplePublicKey@o0.ingest.sentry.io/0',
+ *   debug: nitroApp.h3App.options.debug
+ * })));
+ * ```
+ */
+export const sentryCloudflareNitroPlugin =
+  (optionsOrFn: CloudflareOptions | ((nitroApp: NitroApp) => CloudflareOptions)): NitroAppPlugin =>
+  (nitroApp: NitroApp): void => {
+    nitroApp.localFetch = new Proxy(nitroApp.localFetch, {
+      async apply(handlerTarget, handlerThisArg, handlerArgs: [string, unknown]) {
+        setAsyncLocalStorageAsyncContextStrategy();
+
+        const cloudflareOptions = typeof optionsOrFn === 'function' ? optionsOrFn(nitroApp) : optionsOrFn;
+        const pathname = handlerArgs[0];
+        const event = handlerArgs[1];
+
+        if (!isEventType(event)) {
+          logger.log("Nitro Cloudflare plugin did not detect a Cloudflare event type. Won't patch Cloudflare handler.");
+          return handlerTarget.apply(handlerThisArg, handlerArgs);
+        } else {
+          const requestHandlerOptions = {
+            options: cloudflareOptions,
+            request: { ...event, url: `${event.protocol}//${event.host}${pathname}` },
+            context: event.context.cloudflare.context,
+          };
+
+          return wrapRequestHandler(requestHandlerOptions, () => {
+            const isolationScope = getIsolationScope();
+            const newIsolationScope =
+              isolationScope === getDefaultIsolationScope() ? isolationScope.clone() : isolationScope;
+
+            const traceData = getTraceData();
+            if (traceData && Object.keys(traceData).length > 0) {
+              // Storing trace data in the event context for later use in HTML meta-tags (enables correct connection of parent/child span relationships)
+              // @ts-expect-error Storing a new key in the event context
+              event.context[TRACE_DATA_KEY] = traceData;
```

Comment:
Instead of attaching to `event.context`, can we instead track this via a `WeakMap`?

So we hold a reference to `event.context` or `event`.

Reply:
I saw that `context.cf` is available in both and looks the same. Using a WeakMap now: https://github.com/getsentry/sentry-javascript/pull/15597/commits/a20ebef017a36ec307cfd861cb8c3b8a5cfae2ab